updater/lib/dependabot/update_graph_processor.rb

```diff
@@ -127,13 +129,47 @@ def create_submission(source, files)
 
       GithubApi::DependencySubmission.new(
         job_id: job.id.to_s,
-        # FIXME(brrygrdn): We should obtain the ref from git -or- inject it via the backend service
-        branch: source.branch || "main",
+        branch: branch,
         sha: base_commit_sha,
         package_manager: job.package_manager,
         manifest_file: grapher.relevant_dependency_file,
         resolved_dependencies: grapher.resolved_dependencies
       )
     end
+
+    sig { returns(String) }
+    def default_branch
+      return fetch_default_branch_from_github if job.source.provider == "github"
+
+      Dependabot.logger.warn(
+        "Dependency submissions are not fully supported for provider '#{job.source.provider}'. " \
+        "Substituting 'main' as default branch."
+      )
+      "main"
+    end
+
+    sig { returns(String) }
+    def fetch_default_branch_from_github
+      @fetch_default_branch_from_github ||= T.let(
+        github_client.fetch_default_branch(job.source.repo),
+        T.nilable(String)
+      )
+    rescue Octokit::NotFound
+      # This is unlikely to happen, but if it does it means the repository has been deleted
```

Comment:
I think that this comment may be misleading because this can also happen if the PAT does not have access to the repo

Reply:
Ah, yeah - I'm probably thinking about this in a very specific context we see occasionally where a repository gets deleted when a job is in flight - so we get a 'late' not found after we've gotten past the file fetcher gate for that reason.

I should clarify this by just explaining that in the comment fully.

Reply:
Quick update: we're just going to remove this API call in favour of a simpler git-check: https://github.com/dependabot/dependabot-core/pull/13253

Reply:
Thank you for the explanation and transparency